extensions/amazon-lambda/event-server/src/main/java/io/quarkus/amazon/lambda/runtime/MockEventServer.java

```diff
@@ -0,0 +1,209 @@
+package io.quarkus.amazon.lambda.runtime;
+
+import java.io.Closeable;
+import java.io.IOException;
+import java.util.UUID;
+import java.util.concurrent.BlockingQueue;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.concurrent.Executor;
+import java.util.concurrent.Executors;
+import java.util.concurrent.LinkedBlockingQueue;
+
+import org.jboss.logging.Logger;
+
+import io.vertx.core.Vertx;
+import io.vertx.core.buffer.Buffer;
+import io.vertx.core.http.HttpServer;
+import io.vertx.ext.web.Router;
+import io.vertx.ext.web.RoutingContext;
+import io.vertx.ext.web.handler.BodyHandler;
+
+public class MockEventServer implements Closeable {
+    protected static final Logger log = Logger.getLogger(MockEventServer.class);
+    public static final int PORT = Integer.getInteger("quarkus-internal.aws-lambda.test-port", 8081);
+
+    public void start() {
+        int port = PORT;
+        start(port);
+    }
+
+    public void start(int port) {
+        Vertx vertx = Vertx.vertx();
+        httpServer = vertx.createHttpServer();
+        router = Router.router(vertx);
+        setupRoutes();
+        httpServer.requestHandler(router).listen(port);
```

Comment:
This is an async operation, you need to call .get() to make sure it is actually complete. Otherwise you will get very intermittent test failures where the test starts before the bind operation has actually completed.